azurerm/internal/services/springcloud/spring_cloud_app_data_source.go

```
@@ -0,0 +1,143 @@
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.SpringCloudServiceName,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"https_only": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"is_public": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"persistent_disk": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"mount_path": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"size_in_gb": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceSpringCloudAppRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).AppPlatform.AppsClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := parse.NewSpringCloudAppID(subscriptionId, d.Get("resource_group_name").(string), d.Get("service_name").(string), d.Get("name").(string))
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.AppName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Spring Cloud App %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
```

since this is a Data Source this should raise an error if this doesn't exist - so can we make this:

```suggestion
			return fmt.Errorf("%s was not found", id)
```